src/main/webapp/app/shared/metis/answer-post/answer-post.component.ts

```diff
@@ -66,23 +64,23 @@ export class AnswerPostComponent extends PostingDirective<AnswerPost> implements
     renderer = inject(Renderer2);
     private document = inject<Document>(DOCUMENT);
 
-    @Input() lastReadDate?: dayjs.Dayjs;
-    @Input() isLastAnswer: boolean;
-    @Output() openPostingCreateEditModal = new EventEmitter<void>();
-    @Output() userReferenceClicked = new EventEmitter<string>();
-    @Output() channelReferenceClicked = new EventEmitter<number>();
-    isAnswerPost = true;
+    lastReadDate = input<dayjs.Dayjs | undefined>(undefined);
+    isLastAnswer = input<boolean>(false);
+    isReadOnlyMode = input<boolean>(false);
+    isConsecutive = input<boolean>(false);
```

Comment:
I'm not sure here, but can you double check here too, if you can use `input.required` ?